src/notifier.rs

```diff
@@ -0,0 +1,29 @@
+use tokio::sync::oneshot;
+
+/// StartNotifier offers a convenient way to notify a supervising task that this
+/// task got started or that it failed to start.
+pub struct StartNotifier<T, E>(Box<dyn FnOnce(Result<T, E>) + Send>);
+
+impl<T, E> StartNotifier<T, E> {
+    pub fn from_oneshot(sender: oneshot::Sender<Result<T, E>>) -> Self
+    where
+        T: Send + 'static,
+        E: Send + 'static,
+    {
+        StartNotifier(Box::new(move |err| {
+            let _ = sender.send(err);
```

Comment:
is this `let _` to silence the warning about an unused result? The more typical way to do that is to use the `Result::ok` method which turns it into an `Option`.